ui/src/data/source.tsx

```diff
@@ -1122,3 +1134,18 @@ function saveLocalReviews(reviews: tanagra.ReviewV2[]) {
     })
   );
 }
+
+function parseAPIError<T>(p: Promise<T>) {
```

Comment:
Do the `throw`s here also log to the console? If not, maybe logging the `response` could be helpful for debugging?